runtime/tests/interpreter/transactions_test.go

```diff
@@ -318,3 +321,345 @@ func TestInterpretTransactions(t *testing.T) {
+                      let baz: String
+
+                      prepare(signer: AuthAccount) {
+                          log("a 3")
+                          log(a)
+                          log("b 3")
+                          log(b)
+                          self.baz = signer.address.toString()
+                          log("self.baz")
+                          log(self.baz)
+                      }
+                  }
+
+                  execute {
+                      log("self.foo 2")
+                      log(self.foo)
+                      log("self.role1.bar")
+                      log(self.role1.bar)
+                      log("self.role2.baz")
+                      log(self.role2.baz)
+                  }
+              }
+            `,
+			ParseCheckAndInterpretOptions{
+				CheckerConfig: &sema.Config{
+					BaseValueActivation: baseValueActivation,
+				},
+				Config: &interpreter.Config{
+					BaseActivation: baseActivation,
+				},
+			},
+		)
+		require.NoError(t, err)
+
+		signer := newTestAuthAccountValue(
+			nil,
+			interpreter.AddressValue{0, 0, 0, 0, 0, 0, 0, 1},
+		)
+
+		err = inter.InvokeTransaction(
+			0,
+			interpreter.NewUnmeteredStringValue("A"),
+			interpreter.NewUnmeteredStringValue("B"),
+			signer,
+		)
+		assert.NoError(t, err)
+
+		assert.Equal(t,
+			[]string{
+				// transaction prepare
+				"a 1",
+				"A",
+				"b 1",
+				"B",
+				"self.foo 1",
+				"0x0000000000000001",
+				// role1 prepare
+				"a 2",
+				"A",
+				"b 2",
+				"B",
+				"self.bar",
+				"0x0000000000000001",
+				// role2 prepare
+				"a 3",
+				"A",
+				"b 3",
+				"B",
+				"self.baz",
+				"0x0000000000000001",
+				// execute
+				"self.foo 2",
+				"0x0000000000000001",
+				"self.role1.bar",
+				"0x0000000000000001",
+				"self.role2.baz",
+				"0x0000000000000001",
+			},
+			logs,
+		)
+	})
+
+	t.Run("single role, multiple signers", func(t *testing.T) {
+
+		t.Parallel()
+
+		var logs []string
+
+		valueDeclaration := stdlib.NewStandardLibraryFunction(
+			"log",
+			stdlib.LogFunctionType,
+			"",
+			func(invocation interpreter.Invocation) interpreter.Value {
+				firstArgument := invocation.Arguments[0]
+				message := firstArgument.(*interpreter.StringValue).Str
+				logs = append(logs, message)
+				return interpreter.Void
+			},
+		)
+
+		baseValueActivation := sema.NewVariableActivation(sema.BaseValueActivation)
+		baseValueActivation.DeclareValue(valueDeclaration)
+
+		baseActivation := activations.NewActivation[*interpreter.Variable](nil, interpreter.BaseActivation)
+		interpreter.Declare(baseActivation, valueDeclaration)
+
+		inter, err := parseCheckAndInterpretWithOptions(t,
+			`
+              transaction {
+
+                  prepare(signer1: AuthAccount, signer2: AuthAccount) {}
+
+                  role role1 {
+                      let signer1Address: String
+                      let signer2Address: String
+
+                      prepare(signer1: AuthAccount, signer2: AuthAccount) {
```

Comment:
Should this be allowed? If each signer should have 1 role, we probably need to enforce that each role has one signer and also there are an equal number of roles to signers.